src/VideoCommand.cc

```diff
@@ -350,3 +360,194 @@ Json::Value FindVideo::construct_responses(
     ret[_cmd_name].swap(FindVideo);
     return ret;
 }
+
+//========= FindImage definitions =========
+
+FindFrames::FindFrames() : VideoCommand("FindFrames")
+{
+}
+
+bool FindFrames::get_interval_index (const Json::Value& cmd,
+                                     Json::ArrayIndex& op_index)
+{
+    if (cmd.isMember("operations")) {
+        const auto operations = cmd["operations"];
+        for (auto i = 0; i < operations.size(); i++) {
+            const auto op = operations[i];
+            const std::string& type = get_value<std::string>(op, "type");
+            if (type == "interval") {
+                op_index = i;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+int FindFrames::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    // We try to catch the missing attribute error before
+    // initiating a PMGD query
+    Json::ArrayIndex tmp;
+    bool is_interval = get_interval_index(cmd, tmp);
+    bool is_frames   = cmd.isMember("frames");
+
+    if (!(is_frames != is_interval)) {
```

Comment:
Can you not request the entire video as frames in which case interval could be absent?
I don't follow the logic here of comparing these variables especially with a ! of != instead of ==

Reply:
Shouldn't you instead check if start/stop/step are valid variables and start < stop etc.?

Reply:
This checks that only one_of(interval, frames) is specified. If you just use ==, you need to add an extra condition to make sure at least one exist. The condition as is both checks that at least one is set and that only one is set. 

We can also add checks for start/stop/.

Reply:
About start stop:

actually i would not add a control for that, because if a user wants to the frames and properties, and wrongly set start > stop, we will just return an error when we could have returned properties + empty video. 

This may be a condition we can add in both FindFrames and FindVideo in a future PR. For now we keep the behavior the same as FindVideo.